### PR TITLE
Added basic support for GitHub creators

### PR DESCRIPTION
**Pull request by ErikBjare**
Fixes https://github.com/SuperuserLabs/thankful/issues/46

## TODO

 - [x] Create creators when they don't exist
 - [x] Run the `attributeGithubActivity` sometime appropriate

### CODE REVIEW

File: src/dashboard/DashboardPage.vue [hunk 1]
```diff
@@ -174,6 +171,11 @@ export default {
   created() {
     this.refresh();
   },
+  beforeCreate() {
+    // These below are async, might not have run in time
+    initThankfulTeamCreator();
+    db.attributeGithubActivity();
```
Comment (Powersource) [inline src/dashboard/DashboardPage.vue]: Maybe add .catch() to these to clarify asyncness.

File: src/lib/db.js [hunk 2]
```diff
@@ -179,6 +180,31 @@ export class Database {
...
+      let user_or_org = u.pathname.split('/')[1];
+      if (user_or_org.length > 0) {
+        let creator_url = `https://github.com/${user_or_org}`;
+        await new Creator(creator_url, user_or_org).save();
+        await this.connectActivityToCreator(a.url, creator_url);
+      }
+    });
+
+    await Promise.all(promises);
+    return null;
```
Comment (Powersource) [inline src/lib/db.js]: ?

File: src/lib/db.test.js [hunk 3]
```diff
@@ -12,11 +12,20 @@ Dexie.dependencies.IDBKeyRange = require('fake-indexeddb/lib/FDBKeyRange');
 
 import { Database, Activity, Creator } from './db.js';
 
+async function clearDB(db) {
+  await db.db.creator.clear();
+  await db.db.activity.clear();
```
Comment (Powersource) [inline src/lib/db.test.js]: Might as well clear donation history too?

File: src/lib/db.js (hunk 2 above)
Comment (ErikBjare) [reply src/lib/db.js]: I just don't want to return anything that someone might depend on.

File: src/lib/db.test.js (hunk 3 above)
Comment (ErikBjare) [reply src/lib/db.test.js]: Done

File: src/dashboard/DashboardPage.vue (hunk 1 above)
Comment (ErikBjare) [reply src/dashboard/DashboardPage.vue]: Nah

**Pull request merged**